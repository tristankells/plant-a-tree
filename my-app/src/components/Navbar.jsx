import React, { Component } from "react";

import "../style/Navbar.css";

//Import icons
import AccountCircle from "@material-ui/icons/AccountCircle";
import ShoppingCart from "@material-ui/icons/ShoppingCart";
import Menu from "@material-ui/icons/Menu";

//Import images
import PatLogoText from "../resources/images/textlogo.png";

//Main Navbar component
class Navbar extends Component {

  render() {
    // const navbar = this.props.navbar;
    return (
      <header className="navbar">
        <nav className="navbar__navigation">
          <div className="navbar__burger">
            <ul>
              <li>
                <a>
                  <Menu onMouseDown={this.props.handleBurgerButtonClick} />
                </a>
              </li>
            </ul>
          </div>
          <div className="navbar__logo__container">
            <a href="/">
              <img onMouseDown={this.props.handleLogoCLick} alt="" src={PatLogoText} className="navbar__logo" />
            </a>
          </div>
          <div >
            <p>{this.props.username}</p>
          </div>
          <div className="navbar__links">
            <ul>
              <li>
                <a>
                  <AccountCircle
                    onMouseDown={this.props.handleProfileMenuButtonClick}
                  />
                </a>
              </li>
              <li>
                <a>
                  <ShoppingCart
                    onMouseDown={this.props.handleShoppingCartButtonClick}
                  />
                </a>
              </li>
            </ul>
          </div>
        </nav>
      </header>
    );
  }
}

export default Navbar;
