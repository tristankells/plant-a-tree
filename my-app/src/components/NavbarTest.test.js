import React from "react";
import { shallow } from "enzyme";

import Navbar from "./Navbar";

describe("Navbar", () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Navbar debug />);

    expect(component).toMatchSnapshot();
  });
});
