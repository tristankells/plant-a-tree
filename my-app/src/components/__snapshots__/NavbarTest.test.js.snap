// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render correctly in "debug" mode 1`] = `
<header
  className="navbar"
>
  <nav
    className="navbar__navigation"
  >
    <div
      className="navbar__burger"
    >
      <ul>
        <li>
          <a>
            <pure(MenuIcon) />
          </a>
        </li>
      </ul>
    </div>
    <div
      className="navbar__logo__container"
    >
      <a
        href="/"
      >
        <img
          alt=""
          className="navbar__logo"
          src="textlogo.png"
        />
      </a>
    </div>
    <div>
      <p />
    </div>
    <div
      className="navbar__links"
    >
      <ul>
        <li>
          <a>
            <pure(AccountCircleIcon) />
          </a>
        </li>
        <li>
          <a>
            <pure(ShoppingCartIcon) />
          </a>
        </li>
      </ul>
    </div>
  </nav>
</header>
`;
